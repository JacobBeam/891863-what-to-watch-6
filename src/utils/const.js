export const TabType = {
  OVERVIEW: `Overview`,
  DETAILS: `Details`,
  REVIEWS: `Reviews`
};

export const RatingLevels = {
  BAD: `Bad`,
  NORMAL: `Normal`,
  GOOD: `Good`,
  VERY_GOOD: `Very good`,
  AWESOME: `Awesome`
};
