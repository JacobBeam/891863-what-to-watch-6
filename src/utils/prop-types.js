import PropTypes from 'prop-types';

export const filmsPropTypes = {
  films: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string.isRequired,
    posterImage: PropTypes.string.isRequired,
    previewImage: PropTypes.string.isRequired,
    backgroundImage: PropTypes.string.isRequired,
    backgroundColor: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    runTime: PropTypes.number.isRequired,
    genre: PropTypes.string.isRequired,
    released: PropTypes.number.isRequired,
    id: PropTypes.number.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    videoLink: PropTypes.string.isRequired,
    previewVideoLink: PropTypes.string.isRequired,
  })).isRequired,
};

export const filmPropTypes = {
  film: PropTypes.shape({
    name: PropTypes.string.isRequired,
    posterImage: PropTypes.string.isRequired,
    previewImage: PropTypes.string.isRequired,
    backgroundImage: PropTypes.string.isRequired,
    backgroundColor: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    rating: PropTypes.number.isRequired,
    scoresCount: PropTypes.number.isRequired,
    director: PropTypes.string.isRequired,
    starring: PropTypes.arrayOf(PropTypes.string).isRequired,
    runTime: PropTypes.number.isRequired,
    genre: PropTypes.string.isRequired,
    released: PropTypes.number.isRequired,
    id: PropTypes.number.isRequired,
    isFavorite: PropTypes.bool.isRequired,
    videoLink: PropTypes.string.isRequired,
    previewVideoLink: PropTypes.string.isRequired,
  }).isRequired,
};
