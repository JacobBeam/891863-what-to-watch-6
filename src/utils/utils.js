import {RatingLevels, allGenreFilter} from './const';

export const findFilmById = (films, id) => {
  return films.find((element) => element.id === id);
};


export const conversionMinutes = (mins) => {
  const hours = Math.trunc(mins / 60);
  const minutes = mins % 60;
  return `${hours}h ${minutes}m`;
};


export const findRatingLevel = (rating) => {

  let currentRating = ``;

  switch (true) {
    case (rating >= 0 && rating < 3):
      currentRating = RatingLevels.BAD;
      break;
    case (rating >= 3 && rating < 5):
      currentRating = RatingLevels.NORMAL;
      break;
    case (rating >= 5 && rating < 8):
      currentRating = RatingLevels.GOOD;
      break;
    case (rating >= 5 && rating < 10):
      currentRating = RatingLevels.VERY_GOOD;
      break;
    case (rating >= 10):
      currentRating = RatingLevels.AWESOME;
      break;
  }
  return currentRating;
};

export const filterFilmsByGenre = (genre, films) => {

  if (genre === allGenreFilter.ALL_GENRES) {
    return films;
  }

  return films.filter((film)=>film.genre === genre);
};
