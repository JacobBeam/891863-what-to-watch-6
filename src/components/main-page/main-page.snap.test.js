import React from 'react';
import {render} from '@testing-library/react';
import {createMemoryHistory} from 'history';
import * as redux from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import {Router} from 'react-router-dom';
import MainPage from './main-page';
import {AuthorizationStatus} from '../../utils/const';

const api = jest.fn();
const middlleware = thunk.withExtraArgument(api);
const mockStore = configureStore([middlleware]);
const onFollowingToPlayer = jest.fn();
const onFollowingToMyList = jest.fn();

it(`Should MainPage render correctly`, () => {
  const history = createMemoryHistory();
  const store = mockStore({
    USER: {authorizationStatus: AuthorizationStatus.AUTH},
    FILM_DATA: {films: [{"id": 2, "genre": `Test`}, {"id": 1, "genre": `All`}],
      promo: {
        "genre": `Test`,
        "id": 0,
        "backgroundImage": `Test image`,
        "name": `test name`,
        "poster_image": `Test poster`,
        "released": 2015,
        "isFavorite": false,
      },
      isPromoLoaded: true,
      isFilmsLoaded: true,
      genre: `Test`}
  });

  const {container} = render(
      <redux.Provider store={store}>
        <Router history={history}>
          <MainPage onFollowingToMyList={onFollowingToMyList} onFollowingToPlayer={onFollowingToPlayer}/>
        </Router>
      </redux.Provider>
  );
  expect(container).toMatchSnapshot();
});
