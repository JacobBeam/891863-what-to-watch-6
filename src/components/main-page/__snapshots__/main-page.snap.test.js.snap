// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should MainPage render correctly 1`] = `
<div>
  <section
    class="movie-card"
  >
    <div
      class="movie-card__bg"
    >
      <img
        alt="test name"
        src="Test image"
      />
    </div>
    <h1
      class="visually-hidden"
    >
      WTW
    </h1>
    <header
      class="page-header movie-card__head"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="user-block"
      >
        <div
          class="user-block__avatar"
        >
          <img
            alt="User avatar"
            data-testid="my-list"
            height="63"
            src="img/avatar.jpg"
            style="cursor: pointer;"
            width="63"
          />
        </div>
      </div>
    </header>
    <div
      class="movie-card__wrap"
    >
      <div
        class="movie-card__info"
      >
        <div
          class="movie-card__poster"
        >
          <img
            alt="test name"
            height="327"
            width="218"
          />
        </div>
        <div
          class="movie-card__desc"
        >
          <h2
            class="movie-card__title"
          >
            test name
          </h2>
          <p
            class="movie-card__meta"
          >
            <span
              class="movie-card__genre"
            >
              Test
            </span>
            <span
              class="movie-card__year"
            >
              2015
            </span>
          </p>
          <div
            class="movie-card__buttons"
          >
            <button
              class="btn btn--play movie-card__button"
              data-testid="play"
              type="button"
            >
              <svg
                height="19"
                viewBox="0 0 19 19"
                width="19"
              >
                <use
                  xlink:href="#play-s"
                />
              </svg>
              <span>
                Play
              </span>
            </button>
            <button
              class="btn btn--list movie-card__button"
              data-testid="favorite"
              type="button"
            >
              <svg
                height="20"
                viewBox="0 0 19 20"
                width="19"
              >
                <use
                  xlink:href="#add"
                />
              </svg>
              <span>
                My list
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </section>
  <div
    class="page-content"
  >
    <section
      class="catalog"
    >
      <h2
        class="catalog__title visually-hidden"
      >
        Catalog
      </h2>
      <ul
        class="catalog__genres-list"
      >
        <li
          class="catalog__genres-item "
        >
          <a
            class="catalog__genres-link"
            href="#"
          >
            All genres
          </a>
        </li>
        <li
          class="catalog__genres-item catalog__genres-item--active"
        >
          <a
            class="catalog__genres-link"
            href="#"
          >
            Test
          </a>
        </li>
        <li
          class="catalog__genres-item "
        >
          <a
            class="catalog__genres-link"
            href="#"
          >
            All
          </a>
        </li>
      </ul>
      <div
        class="catalog__movies-list"
      >
        <article
          class="small-movie-card catalog__movies-card"
          data-film-id="2"
        >
          <video
            data-testid="videoplayer"
            height="175"
            width="280"
          />
          <h3
            class="small-movie-card__title"
          >
            <a
              class="small-movie-card__link"
              href="/films/2"
            />
          </h3>
        </article>
      </div>
    </section>
    <footer
      class="page-footer"
    >
      <div
        class="logo"
      >
        <a
          class="logo__link logo__link--light"
          href="#"
        >
          <span
            class="logo__letter logo__letter--1"
          >
            W
          </span>
          <span
            class="logo__letter logo__letter--2"
          >
            T
          </span>
          <span
            class="logo__letter logo__letter--3"
          >
            W
          </span>
        </a>
      </div>
      <div
        class="copyright"
      >
        <p>
          Â© 2019 What to watch Ltd.
        </p>
      </div>
    </footer>
  </div>
</div>
`;
